pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -303,6 +316,30 @@ def newpkgcb(pkgId, name, arch):
             result = await downloader.run()
             metadata_pb.increment()
 
+            repomd_path = result.path
+            repomd = cr.Repomd(repomd_path)
+
+            # Caution: we are not storing when the remote was last updated, so the order of this
+            # logic must remain in this order where we first check the version number as other
+            # changes than sync could have taken place such that the date or repo version will be
+            # different from last sync
+            if (
+                    optimize_sync
+                    and self.remote == self.repository.remote
+                    and (self.repository.last_sync_repo_version
+                         == self.repository.latest_version().number)
+                    and (self.remote.pulp_last_updated
+                         <= self.repository.latest_version().pulp_created)
+                    and repomd.revision == self.repository.revision_number
```

Comment:
@dkliban, @ipanova ,@CodeHeeler , do we want to perform sync if the revision is lower? aka sync old version of the repo?
In pulp2 we didn't I think.

Reply:
I think we want to perform sync when the revision number is higher than the one we have stored.
If optimize_sync is set to True and  revision number is equal or lower then sync should be optimized.
If optimize_sync is set to True and  revision number is higher then full sync  should be done..

Reply:
ok, change made